src/app/getInitialProps.js

```diff
@@ -115,28 +116,32 @@ export async function getAppInitialProps(
     includeSourceIds?: string,
     [key: string]: string,
   },
-  isServer: boolean
+  isServer: boolean,
+  useCache: boolean = true
 ): Promise<AppProps> {
   // flow type is not synced with actual apis
   // $FlowFixMe invalid type definition without userAgentString argument
   const userAgentParser = new UAParser(userAgentString);
   const userAgent = ((userAgentParser.getResult(): any): UAResult);
 
-  const clientSideConfiguration = clientCache.clientSideConfiguration
-    ? clientCache.clientSideConfiguration
-    : await fetchClientSideConfiguration(hostName);
+  const clientSideConfiguration =
+    useCache && clientCache.clientSideConfiguration
+      ? clientCache.clientSideConfiguration
+      : await fetchClientSideConfiguration(hostName);
```

Comment:
1) This way the promises are awaited serially, not in parallel, which might make loading quite slow. `await Promise.all()` would be cool!
2) How about removing the code duplication here:
```javascript
const dataFetchers = {
  clientSideConfiguration: async () => await fetchClientSideConfiguration(hostName),
  translations: () => loadExistingLocalizationByPreference(locales),
  categories: async () => await Categories.generateLookupTables({ locale: preferredLocale });
}

const cachedOrFetchedResults = Object.keys(dataFetchers)
  .map(k => (useCache && clientCache[k]) || dataFetchers[k]())
…
return {
...await Promise.all(cachedOrFetchedResults),
…,
};
...

export function clientStoreAppInitialProps(props: $Shape<AppProps>) {
  Object.keys(dataFetchers).forEach(k => {
    clientCache[k] = props[k] || clientCache[k];
  });
}
```

Reply:
Good catch will add another PR for this.